packages/dds/tree/src/simple-tree/schemaFactory.ts

```diff
@@ -292,6 +293,9 @@ export class SchemaFactory<
 		const Name extends TName,
 		const T extends RestrictiveReadonlyRecord<string, ImplicitFieldSchema>,
 	>(name: Name, fields: T) {
+		// Ensure no collisions between final set of view keys, and final set of stored keys (including those
+		// implicitly derived from view keys)
+		SchemaFactory.assertUniqueKeys(name, fields);
```

Comment:
Note for reviewers: where the uniqueness of view keys is guaranteed by the type system, uniqueness of stored keys is not. This adds runtime validation at when an object schema is created.